### PR TITLE
Internal address support, ssh changes

### PR DESCRIPTION
**Pull request by Secretions**
* Support internal addresses in public rke installs (important for security group restrictions)
* Demo SSH bastion support, works until we get to helm. To get it to 100%, we'd have to either run helm remotely or do an ssh tunnel for it. Rather than do that, I reverted that commit and switched to running all of RanchHand remotely on the bastion. Just pointing out this version for reference in case it does end up being useful. See the commit for details: https://github.com/dominodatalab/ranchhand/commit/2c1aff5c6701238a00b9a9e558cd1fd23eff7241
* In doing the above, I ended up switching to a wrapper around the ssh support that worked out the proxy support. It also abstracted some of the other setup so we don't have to handle that code ourselves. Unsure if you're interested in keeping it, but thought I'd leave it for now so you could evaluate.

All that's really important is the internal address support. Feel free to reimplement, change it, etc.

### CODE REVIEW

File: pkg/ranchhand/run.go [hunk 1]
```diff
@@ -11,11 +11,12 @@ import (
...
 type Config struct {
-	SSHUser    string
-	SSHPort    uint
-	SSHKeyPath string
-	Nodes      []string
-	Timeout    time.Duration
+	SSHUser         string
+	SSHPort         uint
+	SSHKeyPath      string
+	Nodes           []string
```
Comment (sonnysideup) [inline pkg/ranchhand/run.go]: I guess what might have done here is group is the public and private IPs into a single object

```go
type Node struct {
  PublicIP string
  PrivateIP string
}

type Config struct {
  Nodes []Node
}

File: pkg/ssh/client.go [hunk 2]
```diff
@@ -1,58 +1,31 @@
...
-	"github.com/pkg/errors"
-	"golang.org/x/crypto/ssh"
+	"github.com/appleboy/easyssh-proxy"
 )
 
 const Timeout = 5 * time.Second
 
 type Client struct {
-	inner *ssh.Client
+	inner *easyssh.MakeConfig
```
Comment (sonnysideup) [inline pkg/ssh/client.go]: If this does everything we need with less code, then I'm all for it. Good find.

File: cmd/run.go [hunk 3]
```diff
@@ -34,7 +36,8 @@ var (
 )
 
 func init() {
-	runCmd.Flags().StringSliceVarP(&nodeIPs, "node-ips", "n", []string{}, "List of remote hosts")
+	runCmd.Flags().StringSliceVarP(&nodeIPs, "node-ips", "n", []string{}, "Comma-delimited list of remote hosts")
```
Comment (sonnysideup) [inline cmd/run.go]: This is where things get a little sticky. It's pretty dirty to relate public-to-private IPs based on the ordinality of separate lists. We can reformat the way we ingest node IPs (e.g. support `--node-ips "<pub1:priv1>,<pub2:priv2>"` but still allow `--node-ips "<lone1>,<lone2>` ) or move to a file-based config file.

I think we can get by w/o using a config file for now, but we may have to if we end up adding any more complex or compound input values. FWIW, integrating a config file is easy.

**Comment by Secretions** [comment]
> Can you explain how the SSH changes you made support a bastion?

I *reverted* the bastion changes and moved the bastion implementation into the terraform module. See https://github.com/dominodatalab/ranchhand/commit/2c1aff5c6701238a00b9a9e558cd1fd23eff7241 to see the original bastion support, but after getting that far I got to the point of orchestrating helm/kubectl and decided running the whole thing on the bastion was simpler.

File: cmd/run.go (hunk 3 above)
Comment (Secretions) [reply cmd/run.go]: I completely agree, was even going to do exactly that, but a misinterpretation (before I really read through hosts.go, I had the mistaken impression you could pass ip:port,ip:port) got me off that path. I'll switch it to this.

**Pull request closed without merging**